src/qibolab/instruments/rfsoc.py

```diff
@@ -448,40 +543,37 @@ def convert_sweep_results(
         """
         results = {}
 
-        adcs = np.unique([qubits[p.qubit].feedback.ports[0][1] for p in sequence.ro_pulses])
-        for k in range(len(adcs)):
-            for i, serial in enumerate(original_ro):
-                i_pulse = np.array(toti[k][i])
-                q_pulse = np.array(totq[k][i])
-
-                i_vals = i_pulse
-                q_vals = q_pulse
+        adcs = np.unique([qubits[p.qubit].feedback.ports[0][1] for p in original_ro])
+        for k, k_val in enumerate(adcs):
+            adc_ro = [pulse for pulse in original_ro if qubits[pulse.qubit].feedback.ports[0][1] == k_val]
+            for i, ro_pulse in enumerate(adc_ro):
```

Comment:
You could zip the iterators you're using, instead of enumerating one and indexing the others.

It's cleaner, and, in principle, you pay a price (in performances) for indexing a list, higher than looking for the next item.